### ISSUE TITLE
KeyboardHelpDialog extends into nav bar

### ISSUE THREAD

**Issue by zepumph** [phetsims/gravity-force-lab#144]
from https://github.com/phetsims/friction/issues/158

![image](https://user-images.githubusercontent.com/6856943/51578604-88ae4600-1e6a-11e9-94c3-94935465da17.png)

This happens in GFL:B too

**Comment by zepumph** [comment]
I was able to fix this generally by using the same layout strategy that is used in AboutDialog. I factored out AboutDialog's layout strategy into `Dialog.layoutStrategyCenteredInScreen`. 

@pixelzoom please review the refactor since you have done substantial work on Dialog. Note that this differs from how `DEFAULT_LAYOUT_STRATEGY` is set. I'm not sure why the default one isn't set in the statics. My vote may be to move that over to the statics. What do you think?

**Comment by zepumph** [comment]
I just noticed that this solution isn't yet complete. It seems that it needs more duplication from AboutDialog for setting maxHeight:

```
    EXTERNAL_MARGIN  = 12;
. . . 
      // This value was determined empirically and is suitable independent of the ScreenView.layoutBounds.  Scale the
      // entire AboutDialog as a robust solution in case parts outside of the content resize in future versions. Note
      // the layer containing the AboutDialog is already scaled based on the aspect ratio of the window. At a scale of
      // 1.0, the amount of vertical space is 504 and the amount of space above the navigation bar is 464 because the
      // navigation bar takes 40px at that scale.  Please see Sim.resizeAction for more details.
      maxHeight: 464 - EXTERNAL_MARGIN * 2,
```

@pixelzoom in what way do you think would be best to use this value for KeyboardHelpDialog as well?

**Comment by terracoda** [comment]
@zepumph, the dialog no longer overlaps bottom navbar.
Looks really good! Closing.
<img width="838" alt="Screen Shot 2019-08-02 at 2 17 11 PM" src="https://user-images.githubusercontent.com/1902116/62369593-60498780-b530-11e9-923d-82c14dde52b2.png">

**Comment by terracoda** [comment]
Oops, reopening for @pixelzoom to review.

**Comment by pixelzoom** [comment]
> ... @pixelzoom in what way do you think would be best to use this value for KeyboardHelpDialog as well?

This is just bad, it should be based on `ScreenView.DEFAULT_LAYOUT_BOUNDS`.
> `maxHeight: 464 - EXTERNAL_MARGIN * 2`,

Unless there's a reason why we would want a Dialog to be bigger than the ScreenView layoutBounds...  It seems like Dialog should define static constants `DEFAULT_MAX_WIDTH` and `DEFAULT_MAX_HEIGHT` that are based on `ScreenView.DEFAULT_LAYOUT_BOUNDS` with some standard margins.   It might also be good to set Dialog's `maxWidth` and `maxHeight` to these values by default, but that might require dev discussion.

**Comment by zepumph** [comment]
I'm not totally confident on how to progress in this issue. Below are a few items that I think should be independently considered as steps forward. @pixelzoom, for each would you please sign off if you think I should look into doing it. We can also mark for dev meeting if that would be best.

1. Make `layoutStrategyCenteredInScreen` the "DEFAULT_LAYOUT_STRATEGY" in Dialog. This one centers in the screen bounds instead of including the nav bar in the height.
2. Set a default maxHeight based on `phet.joist.sim.screenBoundsProperty`.
3. Export a "DEFAULT_MAX_HEIGHT" to be used elsewhere (where?)
4. Do something with `ScreenView.DEFAULT_LAYOUT_BOUNDS` instead of the actual screen bounds defined in `phet.joist.sim.screenBoundsProperty`

**Comment by pixelzoom** [comment]
> 1. Make layoutStrategyCenteredInScreen the "DEFAULT_LAYOUT_STRATEGY" in Dialog. This one centers in the screen bounds instead of including the nav bar in the height.

Yes.

> 2. Set a default maxHeight based on phet.joist.sim.screenBoundsProperty.

No.  maxWidth and maxHeight should be based on the ScreenView's layoutBounds (the "safe" area) not `screenBoundsProperty` (the full area).   And while this issue has been concerned with maxHeight, the general issue applies to both maxWidth and maxHeight.

> 3.  Export a "DEFAULT_MAX_HEIGHT" to be used elsewhere (where?)

ScreenView layoutBounds is customizable.  So DEFAULT_MAX_HEIGHT (and DEFAULT_MAX_WIDTH) can only be based on the default.  If we want clients to set them explicitly for Dialogs, then make them static constants.  Otherwise no need to make them part of the API.

> 4. Do something with ScreenView.DEFAULT_LAYOUT_BOUNDS instead of the actual screen bounds defined in phet.joist.sim.screenBoundsProperty

Yes, as noted above.

**Comment by zepumph** [comment]
This issue was solved by https://github.com/phetsims/sun/issues/525, out for review there. Closing
